tokio/src/util/blocking_check.rs

```diff
@@ -0,0 +1,25 @@
+#[cfg(unix)]
+use std::os::fd::AsFd;
+
+#[cfg(unix)]
+#[allow(unused_variables)]
+pub(crate) fn check_socket_for_blocking<S: AsFd>(s: &S) -> crate::io::Result<()> {
+    #[cfg(debug_assertions)]
+    {
+        let sock = socket2::SockRef::from(s);
+
+        if !sock.nonblocking()? {
+            eprintln!("Warning: registering a blocking socket, this may be a bug!");
```

Comment:
Is it possible/likely that users are already using stderr for other things? Emitting stderr from within a library feels kinda gross; it might be easier to spot if we just panic here.

Reply:
I actually feel similarly, the request to do it this way specifically came from @carllerche.

I'll leave it up to him to voice why he wants this approach.

Reply:
I spent some more time thinking about this question. I think that I am OK starting with a `debug_assert`. I would like there to be a way to disable the assertion (in case a user has a legitimate case) **, and** the panic message links to an issue for the user to find out how to disable the panic **and** report their legitimate use case.

To disable the panic, we can use a `tokio_allow_from_blocking_fd` `cfg` flag or something similar. The linked issue can include steps for setting it.

cc/ @Darksonn wdyt?

Reply:
Using a debug assertion with a `--cfg` to disable it seems preferable to the `eprintln!`.